cnmodel/util/sound.py

```diff
@@ -495,6 +495,36 @@ def generate(self):
         basenoise = pipnoise(self.time, o['ramp_duration'], o['rate'],
                         o['dbspl'], o['pip_duration'], o['pip_start'], o['seed'])
         return sinusoidal_modulation(self.time, basenoise, o['pip_start'], o['fmod'], o['dmod'], 0.)
+
+
+class ReadWavefile(Sound):
+    def __init__(self, **kwds):
+        reqdWords = ['rate', 'duration', 'dbspl', 'wavefile']
+        for k in reqdWords:
+            if k not in kwds.keys():
+                raise TypeError("ReadWaveFile Missing required argument '%s'" % k)
+        Sound.__init__(self, **kwds)
+        
+    def generate(self):
+        """
+        Read the wave file from disk, and resample if necessary
+        
+        Returns
+        -------
+        array :   generated waveform
+        """
+        [fs_wav, stimulus] = scipy.io.wavfile.read(self.opts['wavefile']) # raw is a numpy array of integer, representing the samples
+        rms = np.sqrt(np.mean(stimulus**2.0))  # find rms of the waveform
```

Comment:
Imagine you wanted to scale the sound such that a specific chunk of the file had a specified dbspl, or if you already know the exact scale factor you want to use. In that case, it would be more work to achieve the level you wanted. Maybe an API like this?
```
dbspl : float or None
    If specified, the wave file is scaled such that its overall dBSPL (measured from RMS) is equal to 
    this value. Either ``dbspl`` or ``scale`` must be specified.
scale : float or None
    If specified, the wave data is multiplied by this value to yield values in dBSPL. 
    Either ``dbspl`` or ``scale`` must be specified.

```